apisix/plugins/ai-rag.lua

```diff
@@ -0,0 +1,158 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local next    = next
+local require = require
+local ngx_req = ngx.req
+
+local http     = require("resty.http")
+local core     = require("apisix.core")
+local decorate = require("apisix.plugins.ai-prompt-decorator").__decorate
+
+local azure_openai_embeddings = require("apisix.plugins.ai-rag.embeddings.azure_openai").schema
+local azure_ai_search_schema = require("apisix.plugins.ai-rag.vector-search.azure_ai_search").schema
+
+local INTERNAL_SERVER_ERROR = ngx.HTTP_INTERNAL_SERVER_ERROR
+local BAD_REQUEST = ngx.HTTP_BAD_REQUEST
+
+local schema = {
+    type = "object",
+    properties = {
+        type = "object",
+        embeddings_provider = {
+            type = "object",
+            properties = {
+                azure_openai = azure_openai_embeddings
+            },
+            -- ensure only one provider can be configured while implementing support for
+            -- other providers
+            required = { "azure_openai" },
+        },
+        vector_search_provider = {
+            type = "object",
+            properties = {
+                azure_ai_search = azure_ai_search_schema
+            },
+            -- ensure only one provider can be configured while implementing support for
+            -- other providers
+            required = { "azure_ai_search" }
+        },
+    },
+    required = { "embeddings_provider", "vector_search_provider" }
+}
+
+local request_schema = {
+    type = "object",
+    properties = {
+        ai_rag = {
+            type = "object",
+            properties = {
+                vector_search = {},
+                embeddings = {},
+            },
+            required = { "vector_search", "embeddings" }
+        }
+    }
+}
+
+local _M = {
+    version = 0.1,
+    priority = 1060, -- TODO check with other ai plugins
```

Comment:
Do we need this comment `TODO check with other ai plugins`?

Reply:
no we don't need this anymore